sacred/observers/file_storage.py

```diff
@@ -51,16 +50,42 @@ def __init__(self, basedir, resource_dir, source_dir, template,
         self.cout = ""
         self.cout_write_cursor = 0
 
-    def queued_event(self, ex_info, command, host_info, queue_time, config,
-                     meta_info, _id):
-        if not os.path.exists(self.basedir):
-            os.makedirs(self.basedir)
+    @staticmethod
+    def _makedirs(name, mode=0o777, exist_ok=False):
+        """ Wrapper of os.makedirs with fallback
+            for exist_ok on python 2.
+        """
+        try:
+            os.makedirs(name, mode, exist_ok)
+        except TypeError:
+            if not os.path.exists(name):
+                os.makedirs(name, mode)
+
+    def _make_run_dir(self, _id):
+        self._makedirs(self.basedir, exist_ok=True)
         if _id is None:
-            self.dir = tempfile.mkdtemp(prefix='run_', dir=self.basedir)
+            for i in range(200):
+                dir_nrs = [int(d) for d in os.listdir(self.basedir)
+                           if os.path.isdir(os.path.join(self.basedir, d)) and
+                           d.isdigit()]
+                _id = max(dir_nrs + [0]) + 1
```

Comment:
I would find this a bit more readable. However, it is not your code, but maybe a good moment to clean it a bit.
```suggestion
                _id = max(*dir_nrs, 0) + 1
```

Reply:
Thank you, good style.

Reply:
Sorry, your patch doesn't work. If `dir_nrs` is a empty list, max cannot handle a single integer.